src/Microsoft.DotNet.Core.Build.Tasks/GenerateRuntimeConfigurationFiles.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.DotNet.ProjectModel;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NuGet.ProjectModel;
+
+namespace Microsoft.DotNet.Cli.Tasks
+{
+    /// <summary>
+    /// Generates the $(project).runtimeconfig.json and optionally $(project).runtimeconfig.dev.json files
+    /// for a project.
+    /// </summary>
+    public class GenerateRuntimeConfigurationFiles : Task
```

Comment:
This isn't the official place for these tasks - they will be in another repo that contains all the "msbuild stuff".  I think it would be best to write the tests then considering that repo will have different infrastructure than what we have in the CLI - it might be a lot of "do over" work if we do it here and then do it again in the new repo.